cluster-configuration-example/cluster-customize-configuration.yaml

```diff
@@ -0,0 +1,213 @@
+  load-balance-ip: IP
+
+  # specify an IP range not in the same network segment with the host machine.
+  service-cluster-ip-range: A.B.C.D/X
+
+
+
+restserver:
+
+  # secret for signing authentication tokens, e.g., "Hello PAI!"
+  jwt_secret: your_jwt_secret
+  # authentication database file path relative to PAI data path
+  lowdb_path: "rest-server/user.db.json"
+  # database admin username
+  lowdb_admin: your_admin_username
+  # database admin password
+  lowdb_passwd: your_admin_password
+
+
+
+hadoop:
+  # If custom_hadoop_binary_path is None, script will download a standard version of hadoop binary for you
+  # hadoop-version
+  # http://archive.apache.org/dist/hadoop/common/hadoop-2.7.2/hadoop-2.7.2.tar.gz
+  custom_hadoop_binary_path: None
+  hadoopversion: 2.7.2
+
+
+
+cluster:
+
+  # HDFS, zookeeper data path on your cluster machine.
+  dataPath: "/datastorage"
+
+  # the docker registry to store docker images that contain system services like frameworklauncher, hadoop, etc.
+  dockerregistryinfo:
+
+    # If public, please fill it the same as your username
+    docker_namespace: your_registry_namespace
+
+    # E.g., gcr.io. If public，fill docker_registry_domain with word "public"
+    # docker_registry_domain: public
+    docker_registry_domain: your_registry_domain
+    # If the docker registry doesn't require authentication, please leave docker_username and docker_password empty
+    docker_username: your_registry_username
+    docker_password: your_registry_password
+
+    docker_tag: your_image_tag
+
+    # The name of the secret in kubernetes will be created in your cluster
+    # Must be lower case, e.g., regsecret.
+    secretname: your_secret_name
+
+
+
+machineinfo:
+
+  NC24R:
+    mem: 224
+    gpu:
+    # type: gpu{type}
+      type: teslak80
+      count: 4
+    cpu:
+      vcore: 24
+    dataFolder: "/mnt"
+    # Note: Up to now, the only supported os version is Ubuntu16.04. Please do not change it here.
+    os: ubuntu16.04
+
+  D8SV3:
+    mem: 32
+    cpu:
+      vcore: 8
+    dataFolder: "/mnt"
+    # Note: Up to now, the only supported os version is Ubuntu16.04. Pls don't change it here.
+    os: ubuntu16.04
+
+
+
+machinelist:
+  # hostname, must be the same as it in the host network
+  infra-01:
+    # The nodename in k8s. To avoid complication, it is suggested to fill in the hostip for the nodename.
+    # If you are confident that DNS can resolve your named nodename correctly, you can specify a UNIQUE_HOST_NAME.
+    # hostip: the static IP of this node. We assume the IP of a host will not change.
+    # sshport: the ssh port of this node. If not specified, the port is 22.
+    # username/password: the user name and password in the OS. Make sure it is granted with sudo permission.
+    hostname: XXXX
+    nodename: IP
+    hostip: IP
+    sshport: PORT
+    etcdid: etcdid1
+    username: username
+    password: password
+    machinetype: D8SV3
```

Comment:
Can we add default sshport, username and password in a common place so that we can simply this 2 lines if username and password are same for the all machines?

Reply:
Done. Set value in DefaultMachineProperties